flaky test: SALChannelTest.testBasicConnectedChannel

in docker (5.2):

```
Test Case 'SALChannelTest.testBasicConnectedChannel' started at 2020-05-11 13:57:55.639
/tmp/.nio-release-tools_AD9w4Z/swift-nio/Tests/NIOTests/SALChannelTests.swift:42: error: SALChannelTest.testBasicConnectedChannel : failed - unexpected syscall write(2147483647, ByteBuffer { readerIndex: 0, writerIndex: 3, readableBytes: 3, capacity: 4, slice: _ByteBufferSlice { 0..<4 }, storage: 0x00007f488811b9d0 (4 bytes) })
<EXPR>:0: error: SALChannelTest.testBasicConnectedChannel : threw error "UnexpectedSyscall(syscall: NIOTests.UserToKernel.write(2147483647, ByteBuffer { readerIndex: 0, writerIndex: 3, readableBytes: 3, capacity: 4, slice: _ByteBufferSlice { 0..<4 }, storage: 0x00007f488811b9d0 (4 bytes) }))"
Test Case 'SALChannelTest.testBasicConnectedChannel' failed (15.017 seconds)
```

--- a/Tests/NIOTests/SyscallAbstractionLayer.swift
+++ b/Tests/NIOTests/SyscallAbstractionLayer.swift
@@ -114,6 +114,18 @@ final class LockedBox<T> {
     }
 }
 
+extension LockedBox where T == UserToKernel {
+    func assertParkedRightNow(file: StaticString = #file, line: UInt = #line) throws {
+        SAL.printIfDebug("\(#function)")
+        let syscall = try self.waitForValue()
+        if case .whenReady = syscall {
+            return
+        } else {
+            XCTFail("unexpected syscall \(syscall)", file: (file), line: line)
+        }
+    }
+}
+
 enum UserToKernel {
     case localAddress
     case remoteAddress
@@ -392,6 +404,9 @@ extension HookedSelector {
         try self.wakeups.takeValue()
     }
 
+    func assertParkedRightNow(file: StaticString = #file, line: UInt = #line) throws {
+        try self.userToKernel.assertParkedRightNow(file: file, line: line)
+    }
 }
 
 extension EventLoop {
@@ -410,6 +425,7 @@ extension EventLoop {
         }
         try hookedSelector.assertWakeup(file: (file), line: line)
         try syscallAssertions()
+        try hookedSelector.userToKernel.assertParkedRightNow() // We need the EventLoop to finish running its tick.
         return try box.takeValue().get()
     }
 }
@@ -434,6 +450,7 @@ extension EventLoopFuture {
             }
         }
         try hookedSelector.assertWakeup()
+        try hookedSelector.userToKernel.assertParkedRightNow() // We need the EventLoop to finish running its tick.
         return try box.takeValue().get()
     }
 }
@@ -585,13 +602,7 @@ extension SALTest {
     }
 
     func assertParkedRightNow(file: StaticString = #file, line: UInt = #line) throws {
-        SAL.printIfDebug("\(#function)")
-        let syscall = try self.userToKernelBox.waitForValue()
-        if case .whenReady = syscall {
-            return
-        } else {
-            XCTFail("unexpected syscall \(syscall)", file: (file), line: line)
-        }
+        try self.userToKernelBox.assertParkedRightNow(file: file, line: line)
     }
 
     func assertWaitingForNotification(result: SelectorEvent<NIORegistration>?,

